libs/langchain-google-common/src/llms.ts

```diff
@@ -207,7 +244,44 @@ export abstract class GoogleBaseLLM<AuthOptions>
       {},
       options as BaseLanguageModelCallOptions
     );
-    const ret = responseToBaseMessage(result);
+    const ret = safeResponseToBaseMessage(result, this.safetyHandler);
     return ret;
   }
+
+  createProxyChat(): ChatGoogleBase<AuthOptions> {
+    console.log("this", this);
+    return new ProxyChatGoogle<AuthOptions>({
+      ...this.originalFields,
+      connection: this.connection,
+    });
+  }
+
+  async invoke(
+    input: BaseLanguageModelInput,
+    options?: BaseLanguageModelCallOptions
+  ): Promise<string> {
+    const proxyChat = this.createProxyChat();
+    const chunk = await proxyChat.invoke(input, options);
+    return chunkToString(chunk);
+  }
+
+  async stream(
+    input: BaseLanguageModelInput,
+    options?: Partial<BaseLanguageModelCallOptions>
+  ): Promise<IterableReadableStream<string>> {
+    const proxyChat = this.createProxyChat();
+
+    const chatStream = await proxyChat.stream(input, options);
+
+    // Based on code from Gemini Advanced
+    const transformStream = new TransformStream({
+      transform(chunk: BaseMessageChunk, controller) {
+        const text = chunkToString(chunk);
+        controller.enqueue(text);
+      },
+    });
+
+    const readableStream = chatStream.pipeThrough(transformStream);
+    return IterableReadableStream.fromReadableStream(readableStream);
```

Comment:
I think this will break the callback system - there's a few behind the scenes things going on. We'll want to override `_streamResponseChunk` instead.

Reply:
The problem with overriding `_streamResponseChunk()` is that it **requires** a string `_prompt` while `stream()` takes a `BaseLanguageModelInput` which will allow sending a multimodal prompt to the LLM, which is part of why we're doing this.

I suppose we could change `_streamResponseChunk()` _in addition to_ adding this `stream()` method?

What are we breaking in callbacks by overriding `stream()`?

Reply:
The underlying problem is that `BaseLLM._streamIterator()` explicitly calls the static method `BaseLLM._convertInputToPromptValue()` which forces the prompt to be a string.

I guess we could override `_streamIterator()`, but that is a huge method to duplicate the code for. Unless it is safe to have our method call the chat model version of `_streamIterator()`?

Reply:
I've gone with overriding `_streamIterator()` see how that does?
(It passes all the tests I have for google-common, but I'm not sure what other tests there are that might be testing out stream().)